File: frontend/src/types/medication.ts
export interface Medication {
	id: number
	name: string
	form: string
	price: number
	quantity: number
	activeSubstance: string
	expirationDate: string
}

export interface MedicationInPharmacy {
	id: number
	name: string
	form: string
	price: number
	quantity: number
	activeSubstance: string
	expirationDate: string
	pharmacyId: number
}

File: pharmacy/src/main/java/ru/ssu/pharmacy/entity/Medication.java
package ru.ssu.pharmacy.entity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
@Entity
@Table(name = "medication")
@Getter @Setter @NoArgsConstructor
public class Medication {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String medicationName;
    private String form;
    private String medicationDescription;

    @ManyToOne
    @JoinColumn(name = "active_substance_id", nullable = false)
    private ActiveSubstance activeSubstance;
}

